apps/vthread_props/src/main/java/vthread_props/Main.java

```diff
@@ -0,0 +1,223 @@
+        final AtomicReference<Exception> exception = new AtomicReference<>();
+        final Thread vt = Thread.ofVirtual().start(() -> {
+            try {
+                System.out.println("Virtual: " + (Thread.currentThread().isVirtual() ? "True" : "ERROR"));
+                final LocalDateTime now = LocalDateTime.now();
+                dateTimeRef.set(now);
+                System.out.println("Time: " + now.format(DateTimeFormatter.ISO_LOCAL_DATE_TIME));
+            } catch (Exception e) {
+                exception.set(e);
+                e.printStackTrace();
+            } finally {
+                l.countDown();
+            }
+        });
+        l.await(1, TimeUnit.MICROSECONDS);
+        vt.join();
+        if (exception.get() != null) {
+            throw exception.get();
+        }
+        return dateTimeRef.get() != null;
+    }
+
+    private static boolean testZoneIdAccess() throws Exception {
+        final CountDownLatch l = new CountDownLatch(1);
+        final AtomicReference<ZoneId> zoneIdRef = new AtomicReference<>();
+        final AtomicReference<Exception> exception = new AtomicReference<>();
+        final Thread vt = Thread.ofVirtual().start(() -> {
+            try {
+                System.out.println("Virtual: " + (Thread.currentThread().isVirtual() ? "True" : "ERROR"));
+                ZoneId zoneId = ZoneId.systemDefault();
+                zoneIdRef.set(zoneId);
+                System.out.println("System default zone: " + zoneId);
+            } catch (Exception e) {
+                exception.set(e);
+                e.printStackTrace();
+            } finally {
+                l.countDown();
+            }
+        });
+        l.await(1, TimeUnit.MICROSECONDS);
+        vt.join();
+        if (exception.get() != null) {
+            throw exception.get();
+        }
+        return zoneIdRef.get() != null;
+    }
+
+    private static boolean testMultipleTimeOperations() throws Exception {
+        final int threadCount = 10;
+        final CountDownLatch l = new CountDownLatch(threadCount);
+        final AtomicReference<Exception> exception = new AtomicReference<>();
+        try (ExecutorService executor = Executors.newVirtualThreadPerTaskExecutor()) {
+            for (int i = 0; i < threadCount; i++) {
+                final int threadNum = i;
+                executor.submit(() -> {
+                    try {
+                        System.out.println("Thread " + threadNum + " is virtual: " + (Thread.currentThread().isVirtual() ? "True" : "ERROR"));
+                        System.out.println("Thread " + threadNum + ": " + LocalDateTime.now() + " in zone " + ZoneId.systemDefault());
+                    } catch (Exception e) {
+                        exception.set(e);
+                        e.printStackTrace();
+                    } finally {
+                        l.countDown();
+                    }
+                });
+            }
+            l.await(1, TimeUnit.MICROSECONDS);
```

Comment:
Not checking `await`'s return value and setting the timeout to 1 microseconds seems a bit "risky". I would expect the test to fail in case of a timeout.

Reply:
Thanks, this is me playing with it and leaving it half-there. Fixing...